gapic-generator/templates/default/service/client/_config.text.erb

```diff
@@ -47,6 +45,25 @@
 #    *  (`GRPC::Core::ChannelCredentials`) a gRPC credentails object
 #    *  (`nil`) indicating no credentials
 #
+#   @deprecated Passing a `String` to a keyfile path or a `Hash` of credentials
```

Comment:
This `@deprecated` will deprecate the whole `credentials` attribute so we should not do that.
Use `@note` and add "Warning: ". While we're at it the "Warning" about "accepting a credentials configuration" below can be a `@note` as well.